programs/mango-v4/src/instructions/account_settle_fees_accrued_with_mngo.rs

```diff
@@ -0,0 +1,98 @@
+use anchor_lang::prelude::*;
+use checked_math as cm;
+use fixed::types::I80F48;
+
+use crate::accounts_zerocopy::*;
+use crate::error::MangoError;
+use crate::state::*;
+
+use crate::accounts_ix::*;
+
+pub fn account_settle_fees_accrued_with_mngo(
+    ctx: Context<AccountSettleFeesAccruedWithMngo>,
+    max_settle: u64,
+) -> Result<()> {
+    // Cannot settle with yourself
+    require_keys_neq!(
+        ctx.accounts.account.key(),
+        ctx.accounts.dao_account.key(),
+        MangoError::CannotSettleWithSelf
+    );
+
+    let group = ctx.accounts.group.load()?;
+
+    let mut mngo_bank = ctx.accounts.mngo_bank.load_mut()?;
+    let mut settle_bank = ctx.accounts.settle_bank.load_mut()?;
+
+    let mut account = ctx.accounts.account.load_full_mut()?;
+    // account constraint #1
+    require!(
+        account.fixed.is_owner_or_delegate(ctx.accounts.owner.key()),
+        MangoError::SomeError
+    );
+    let mut dao_account = ctx.accounts.dao_account.load_full_mut()?;
+
+    // check positions exist, for nicer error messages
```

Comment:
I'd use enforce_token_position here. Maybe the USDC position on the dao_account or user account got emptied and implicitly closed.

The current code makes sense for settle_pnl because the USDC position is marked as "in_use" for the time that the user has open perp positions.